core/kazoo_modb-1.0.0/priv/couchdb/views/cdrs.json

```diff
@@ -5,6 +5,17 @@
         "crossbar_listing": {
             "map": "function(doc) {if (doc.pvt_deleted || doc.pvt_type != 'cdr') return;emit(doc.pvt_created, {id: doc._id,call_id: doc.call_id,caller_id_name: doc.caller_id_name,caller_id_number: doc.caller_id_number,callee_id_name: doc.callee_id_name,callee_id_number: doc.callee_id_number,duration_seconds: doc.duration_seconds,billing_seconds: doc.billing_seconds,timestamp: doc.timestamp,hangup_cause: doc.hangup_cause,other_leg_call_id: doc.other_leg_call_id,owner_id: doc.custom_channel_vars.owner_id,to: doc.to_uri,from: doc.from_uri,inception: doc.custom_channel_vars.inception,direction: doc.call_direction,request: doc.request,authorizing_id: doc.custom_channel_vars.authorizing_id});}"
         },
+        "group_listing": {
+            "map": "function(doc) {if (doc.pvt_deleted || doc.pvt_type != 'cdr') return; emit([doc.group_time, doc.group_key, doc.channel_created_time], {id: doc._id, channel_time: doc.channel_created_time});}",
+            "reduce": "function(keys, values, rereduce) {    if (rereduce) { return values.reduce(function(a, b) { return a == null ? b : b == null ? a : a.channel_time < b.channel_time ? a : b }) } else { v = values.reduce(function(a, b) { return a == null ? b : b == null ? a : a.channel_time < b.channel_time ? a : b }); return v; }}"
+        },
+        "group_listing_by_id": {
+            "map": "function(doc) { if(doc.pvt_type != 'cdr' || doc.pvt_deleted) return; emit([doc.group_id, doc.channel_created_time], null)}"
+        },
+        "group_listing_by_owner": {
+            "map": "function(doc) {if (doc.pvt_deleted || doc.pvt_type != 'cdr' || !doc.custom_channel_vars || !doc.custom_channel_vars.owner_id) return; emit([doc.custom_channel_vars.owner_id, doc.group_time, doc.group_key, doc.channel_created_time], {id: doc._id,  channel_time: doc.channel_created_time});}",
+            "reduce": "function(keys, values, rereduce) {    if (rereduce) { return values.reduce(function(a, b) { return a == null ? b : b == null ? a : a.channel_time < b.channel_time ? a : b }) } else { v = values.reduce(function(a, b) { return a == null ? b : b == null ? a : a.channel_time < b.channel_time ? a : b }); return v; }}"
+        },
```

Comment:
Please brace your nested tertiary conditionals or use ifs.

Curious if there would be a way to be even less readable than that.
And are `\n`s that expensive nowadays?